### DISCUSSION TITLE
Failure after termux arm64 upgrade

### DISCUSSION THREAD

**Discussion by ddkwork** [fyne-io/fyne#3421] [General]
sudo pacman -S go xorg-server-devel libxcursor libxrandr libxinerama libxi

![Screenshot_20221122_140424_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/203237492-4d53ca98-43a0-441b-9bcd-e69da8836364.jpg)

**Comment by ddkwork** [comment]
arch linux latest tested

**Comment by andydotxyz** [comment]
Can you please share what device  this is? It does not look like a standard arm64 linux computer

**Comment by ddkwork** [comment]
> Can you please share what device this is? It does not look like a standard arm64 linux computer

![Screenshot_20221122_231011_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/203349683-88d5f9bd-0363-4a15-8d1d-4cf5e08d5bec.jpg)

wubantu also not work

**Comment by ddkwork** [comment]
> Can you please share what device  this is? It does not look like a standard arm64 linux computer

termux proot linux

**Comment by ddkwork** [comment]
it seem like the opengl version is wrong.

I was tested 3 linuxs:
Wubantu,arch,manjaro.

Fyne was working Some days ago, some one  day not work when exec command "apt upgrade".

**Comment by andydotxyz** [comment]
It sounds like your upgrade somehow removed or disabled the MIT-SHM X11 extension

**Comment by ddkwork** [comment]
> It sounds like your upgrade somehow removed or disabled the MIT-SHM X11 extension

Hello you, what i should apt install?

**Comment by ddkwork** [comment]
Nothing search found.

![Screenshot_20221122_235137_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/203359755-cb0f4bf3-f56c-492e-bcf7-b293116d69d2.jpg)

**Comment by andydotxyz** [comment]
I don't know - I have never worked with that combination.
All I can suggest is you go through the setup instructions again so all the right libraries are installed.

**Comment by ddkwork** [comment]
> I don't know - I have never worked with that combination.
> All I can suggest is you go through the setup instructions again so all the right libraries are installed.

I was reinstall system and exec this:

sudo pacman -S go xorg-server-devel libxcursor libxrandr libxinerama libxi

sudo apt-get install golang gcc libgl1-mesa-dev xorg-dev

But not working.

**Comment by andydotxyz** [comment]
Is it exactly the same error? I see in your previous messages errors about `gl` missing as well, so it's not entirely clear at what stage things are failing...

**Comment by ddkwork** [comment]
> Is it exactly the same error? I see in your previous messages errors about `gl` missing as well, so it's not entirely clear at what stage things are failing...

gioui was working with Wayland.

**Reply by andydotxyz** [reply]
You can test Fyne wayland support (not yet on by default) using go run -tags wayland ..
If your platform is not X this may help a lot!

**Reply by ddkwork** [reply]
> You can test Fyne wayland support (not yet on by default) using go run -tags wayland ..
> If your platform is not X this may help a lot!

![Screenshot_20221129_213256_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/204542388-3bac54d4-27be-441e-9cc3-7865a5b62ebb.jpg)

Fyne still not working use wayland, gio was working. But i love fyne only.

I was input this command :

go run . -tags wayland

**Reply by andydotxyz** [reply]
You are not managing to run the wayland tags. Your file name is in the wrong position.

`go run -tags wayland main.go` - apologies if my earlier message was not clear

**Reply by ddkwork** [reply]
> You are not managing to run the wayland tags. Your file name is in the wrong position.
> 
> `go run -tags wayland main.go` - apologies if my earlier message was not clear

![Screenshot_20221129_224217_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/204559238-fd7b92ca-4a50-4bb3-b5c6-d67629277693.jpg)

**Reply by andydotxyz** [reply]
Thanks, it's not moved to Wayland then ;)

**Comment by Bluebugs** [comment]
If I remember correctly, the x11 shm extension is for a software backend. Something mesa would fallback to if the gpu driver isn't installed. As you are mentioning arm and also Wayland, what is your gpu and are you using x11 or Wayland for the display system?

**Comment by ddkwork** [comment]
> If I remember correctly, the x11 shm extension is for a software backend. Something mesa would fallback to if the gpu driver isn't installed. As you are mentioning arm and also Wayland, what is your gpu and are you using x11 or Wayland for the display system?

X11, 10 days ago fyne is working.

This system is based on Android termux proot container， seem like docker.

**Reply by andydotxyz** [reply]
Are you certain it is still X11? It is possible the system upgrade changed things...
Can you run something like `Xeyes` to test?

**Reply by ddkwork** [reply]
> Are you certain it is still X11? It is possible the system upgrade changed things...
> Can you run something like `Xeyes` to test?

Xeyes ？ how to do？

**Reply by ddkwork** [reply]
> > Are you certain it is still X11? It is possible the system upgrade changed things...
> > Can you run something like `Xeyes` to test?
> 
> Xeyes ？ how to do？

![Screenshot_20221129_225133_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/204561621-e9aa16b2-8689-4a83-9be5-e11f6be8f72b.jpg)

not found

**Reply by ddkwork** [reply]
> > Are you certain it is still X11? It is possible the system upgrade changed things...
> > Can you run something like `Xeyes` to test?
> 
> Xeyes ？ how to do？

yes， x11 still, used x11 vnc connet it.

**Reply by andydotxyz** [reply]
> > Are you certain it is still X11? It is possible the system upgrade changed things...
> > Can you run something like `Xeyes` to test?
> 
> Xeyes ？ how to do？

Sorry, we are not able to help you work with your package manager - no idea how termux searches and installs software.

**Comment by andydotxyz** [comment]
I would recommend:

* remove all Go cache
* update system packages
* reboot
* compile again

According to some other internet posts this has helped other apps - as there seems to be a mismatch between your X, the memory manager, and the app binary...

**Comment by ddkwork** [comment]
> I would recommend:
> 
> * remove all Go cache
> * update system packages
> * reboot
> * compile again
> 
> According to some other internet posts this has helped other apps - as there seems to be a mismatch between your X, the memory manager, and the app binary...

Ok, wait me.

**Reply by ddkwork** [reply]
![Screenshot_20221129_230423_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/204566178-646650fa-f36a-4711-88e9-d5c320e64e25.jpg)

![Screenshot_20221129_230849_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/204566204-9b58c3ba-c22c-4a33-93ac-922ac441969d.jpg)

Still not working with delete all cache.

**Reply by andydotxyz** [reply]
I don't think that is where Go keeps it's package cache - try `go clean -cache -modcache -i -r`.
Did you do all the other steps as well?
No point in using the `wayland` tag any more, we know it doesn't work now.

**Reply by ddkwork** [reply]
> I don't think that is where Go keeps it's package cache - try `go clean -cache -modcache -i -r`.
> Did you do all the other steps as well?
> No point in using the `wayland` tag any more, we know it doesn't work now.

I was delete all go build cache ,mod cache ,reboot system etc, but hello word not run.

**Reply by andydotxyz** [reply]
Is it still the same error?

**Reply by ddkwork** [reply]
> Is it still the same error?

yes

**Reply by ddkwork** [reply]
> Is it still the same error?

Hi,what gles version need change?

![Screenshot_20221215_213956_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/207873771-6f1583f9-d224-4293-a7ff-645fd1e4a253.jpg)

**Reply by ddkwork** [reply]
![Screenshot_20221219_205657_com.realvnc.viewer.android.jpg](https://user-images.githubusercontent.com/19886504/208431256-a1294b44-f0b4-40dd-a10f-7133fd9c4e7b.jpg)

working NOW，please set version in install commands or make a glfw build  script from src with vcs bindding.

**Reply by andydotxyz** [reply]
It is not clear what command you issues to fix this and/or why it worked - the OpenGL information you posted seems to be consistent.

**Reply by ddkwork** [reply]
> It is not clear what command you issues to fix this and/or why it worked - the OpenGL information you posted seems to be consistent.

mesa 22.2.3 update to 22.3.1
compare two picture

all in one steps is update all x11 about，i am sure that problem is .o file  version number check.

The error info serial number 209 and need 210，i don't how to get this value from glfw，but  i believe it will be return from .o file，so we need add version check.

**Reply by ddkwork** [reply]
> It is not clear what command you issues to fix this and/or why it worked - the OpenGL information you posted seems to be consistent.

so，if we add version check in this command will be better.

sudo pacman -S go xorg-server-devel libxcursor libxrandr libxinerama libxi

**Reply by andydotxyz** [reply]
If no version numbers were specified I wonder why your system did not have the latest versions installed? 
If the issue was "upgrade to latest packages" then I don't think there is an issue here

**Reply by ddkwork** [reply]
> If no version numbers were specified I wonder why your system did not have the latest versions installed? 
> If the issue was "upgrade to latest packages" then I don't think there is an issue here

I'm a little confused. Is there really such a big gap between Chinese and English? If I can solve this problem by updating the version when I publish the discussion, will I still discuss it here? Mesa just released it a few days ago, and I can only provide the solution here according to the process I handled. As for whether you will fix this kind of problem, it's up to you.

**Reply by andydotxyz** [reply]
It sounds like you solved it by upgrading. 
So I don't know what we can do to fix it...
Our documentation does not specify install versions, so they should always be latest.
What else can we specify?

**Comment by Bluebugs** [comment]
I would like to point out that all the output returned by glxinfo indicate a software renderer and not a GPU based one. This might create some problem.
